lib/widgets/pip/pip_app.dart

```diff
@@ -0,0 +1,172 @@
+      canPop: pipController.isPipShowing,
+      child: Scaffold(
+        body: PipContainer(
+            content: widget.child ?? const SizedBox(),
+            controller: pipController,
+            pipStateBuilder: (_, state) {
+              switch (state) {
+                case PipState.full:
+                  return _buildWebView(context);
+                case PipState.minimize:
+                  return Stack(
+                    children: [
+                      _buildWebView(context),
+                      Positioned(
+                        right: 0,
+                        top: 16,
+                        child: IconButton(
+                          onPressed: () {
+                            runJavaScript('window.player.unminimize()');
+                            pipController.full();
+                          },
+                          icon: const Icon(
+                            Icons.expand,
+                          ),
+                        ),
+                      ),
+                    ],
+                  );
+
+                default:
+                  return const SizedBox();
+              }
+            }),
+      ),
+    );
+  }
+
+  void initWebController() {
+    late final PlatformWebViewControllerCreationParams params;
+    if (WebViewPlatform.instance is WebKitWebViewPlatform) {
+      params = WebKitWebViewControllerCreationParams(
+        allowsInlineMediaPlayback: true,
+        mediaTypesRequiringUserAction: const <PlaybackMediaTypes>{},
+      );
+    } else {
+      params = const PlatformWebViewControllerCreationParams();
+    }
+    final WebViewController controller =
+        WebViewController.fromPlatformCreationParams(params);
+    controller
+      ..setJavaScriptMode(JavaScriptMode.unrestricted)
+      ..setNavigationDelegate(
+        NavigationDelegate(),
+      )
+      ..addJavaScriptChannel(
+        'LoraChannel',
+        onMessageReceived: (JavaScriptMessage message) async {
+          Map<String, dynamic> jsonMap = json.decode(message.message);
+          final payload = jsonMap.containsKey('payload')
+              ? jsonMap['payload'] as Map<String, dynamic>
+              : null;
+
+          switch (jsonMap['eventName'] as String) {
+            case 'player.INITIALIZE':
+              runJavaScript('window.player.open("$_showId")');
+              break;
+            case 'player.SHOW_PRODUCT_VIEW':
+              debugPrint(message.message);
+              minimize();
+              // await Future.delayed(const Duration(milliseconds: 200));
+              navigatorKey.currentState?.push(
+                MaterialPageRoute(
+                    builder: (_) =>
+                        ProductDetailScreen(sku: 'sku', title: 'title')),
+              );
+
+              break;
+            case 'player.MINIMIZED':
+              minimize();
+              break;
+            case 'player.CLOSE':
+              close();
+              break;
+          }
```

Comment:
Could we reuse these codes by using `webview_screen.dart`? Or re-write another widget so that we don't repeat this.